### PR TITLE
Fix _call unicode issues

### LINKED ISSUE #488: Fix leapp.stdlib.call
Problem is around the `callback_raw` and `callback_linebuffered` functions. Documentation in the `_call` function tells (a), implementation inside `_multiplex` tells (b) (and in advance, the implementation is broken) and our callback functions in `leapp.stdlib` are kinda logic (c).

tl;dr - the current state is not systematic. Someone should sit above this and should fix the current state so the logic, doc, naming, ... will be clean and reliable.

PLUS: we need tests! Whoever will take this, please think about the implementation and design before you implement this. Keep in mind that some `callback_raw` functions are implemented in some actors as well, so when signature of the `callback_raw` function is changed, the modification will have to appear in such actors as well.

Hard requirement: __all related functions are really covered by tests!!__

### PR DESCRIPTION
**Pull request by shaded-enmity**
* Input data are now decoded using incremental decoder
* Docstring now reflects parameters better
* Add tests for various language pangrams

Fixes #488

### CODE REVIEW

**Comment by shaded-enmity** [comment]
Note that #481 should be merged first then I'll rebase. I've accidentally squashed Jozef's commit :angel:

**Comment by bocekm** [comment]
ping @drehak - you can test if this fixes also the returned non-unicode stuff you've dealt with today.

File: leapp/libraries/stdlib/call.py [hunk 1]
```diff
@@ -68,21 +65,19 @@ def _multiplex(ep, read_fds, callback_raw, callback_linebuffered,
...
-    #         # [stdout, stderr] is relayed, stdout=1 a stderr=2
-    #         # as the field starting indexed is 0, so the +1 needs to be added
-    #         callback_linebuffered(read_fds.index(fd) + 1, lb)
+    # Process leftovers from line buffering
+    if encoding:
+        for (fd, lb) in linebufs.items():
+            if lb:
+                # [stdout, stderr] is relayed, stdout=1 a stderr=2
+                # as the field starting indexed is 0, so the +1 needs to be added
+                callback_linebuffered(read_fds.index(fd) + 1, lb)
```
Comment (pirat89) [inline leapp/libraries/stdlib/call.py]: Still broken :)

```suggestion
                callback_linebuffered((fd, read_fds.index(fd) + 1), lb)
```

Comment (pirat89) [reply leapp/libraries/stdlib/call.py]: Annyway, `red_fds.index(fd) + 1` is several times (2x seeing now). From the point that this should provide fd_type, what about rather write nested function `_get_fd_type`, like:
```
        def _get_fd_type(fd):
            return read_fds.index(fd) + 1
```

File: tests/scripts/test_stdlib_call.py [hunk 2]
```diff
@@ -29,14 +31,14 @@ def test_stdin_fd():
...
-#
-#     def callback(fd, data):
-#         buffered.append(data)
-#     _call(('bash', '-c', 'echo 1; echo 2; echo 3'), callback_linebuffered=callback)
-#     assert buffered == ['1', '2', '3']
+
+def test_linebuffer_callback():
+    buffered = []
+
+    def callback(fd, data):
```
Comment (pirat89) [inline tests/scripts/test_stdlib_call.py]: just nitpick, fd_info would be better, but I am ok with fd as well as it is not used.

**Comment by pirat89** [comment]
close the PR and create new one with "correct" branch name :) - or add fix to Makefile
 - our sed doesn't like `/`

**Comment by shaded-enmity** [comment]
Even after updating `.flake8` to ignore E117 it still reports E117. At this point I'd appreciate someone looking into our CI setup. I don't think a 2h game of whack-a-mole with misconfigured linter is worth anyones time.

File: leapp/libraries/stdlib/call.py [hunk 3]
```diff
@@ -94,25 +97,35 @@ def _call(command, callback_raw=lambda fd, value: None, callback_linebuffered=la
...
         raise TypeError('command parameter has to be a list or tuple')
-    if not callable(callback_raw) or callback_raw.__code__.co_argcount != 2:
-        raise TypeError('callback_raw parameter has to be callable accepting 2 parameters')
-    if not callable(callback_linebuffered) or callback_linebuffered.__code__.co_argcount != 2:
+    if not callable(callback_raw) or\
+            (getattr(callback_raw, '__code__', None) and callback_raw.__code__.co_argcount != 2):
+            raise TypeError('callback_raw parameter has to be callable accepting 2 parameters')
+    if not callable(callback_linebuffered)\
+            or (getattr(callback_linebuffered, '__code__', None)
+                and callback_linebuffered.__code__.co_argcount != 2):
```
Comment (jmikovic) [inline leapp/libraries/stdlib/call.py]:
```suggestion
                and callback_linebuffered.__code__.co_argcount != 2): # noqa
```

should ignore flake8 error

File: leapp/libraries/stdlib/call.py [hunk 4]
```diff
@@ -94,25 +97,35 @@ def _call(command, callback_raw=lambda fd, value: None, callback_linebuffered=la
...
         :rtype: dict
     """
     if not isinstance(command, (list, tuple)):
         raise TypeError('command parameter has to be a list or tuple')
-    if not callable(callback_raw) or callback_raw.__code__.co_argcount != 2:
-        raise TypeError('callback_raw parameter has to be callable accepting 2 parameters')
-    if not callable(callback_linebuffered) or callback_linebuffered.__code__.co_argcount != 2:
+    if not callable(callback_raw) or\
+            (getattr(callback_raw, '__code__', None) and callback_raw.__code__.co_argcount != 2):
+            raise TypeError('callback_raw parameter has to be callable accepting 2 parameters')
```
Comment (jmikovic) [inline leapp/libraries/stdlib/call.py]:
```suggestion
        raise TypeError('callback_raw parameter has to be callable accepting 2 parameters')
```

fixes over-indent error

**Comment by shaded-enmity** [comment]
Thanks to @jmikovic this PR now passes all the flake8 hurdles!

**Pull request merged**